day2/exercise2/solution.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+func allocate(noOfStudents int, ratings chan int) {
+	for i := 0; i < noOfStudents; i++ {
+		time.Sleep(time.Duration(rand.Intn(100)) * time.Millisecond)
+		rating := rand.Intn(10) + 1
+		ratings <- rating
+	}
+	close(ratings)
+}
+
```

Comment:
You have to take ratings from each classmate concurrently.

Reply:
You are finding the ratings synchronously, although you are using goroutine.

Reply:
yes thanks, I have added the suggested changes.